middleware/kubernetes/kubernetes.go

```diff
@@ -45,15 +45,39 @@ type Kubernetes struct {
 	Selector      *labels.Selector
 }
 
+type endpoint struct {
+	addr api.EndpointAddress
+	port api.EndpointPort
+}
+
+type service struct {
+	name      string
+	namespace string
+	addr      string
+	ports     []api.ServicePort
+	endpoints []endpoint
+}
+
 var errNoItems = errors.New("no items found")
 var errNsNotExposed = errors.New("namespace is not exposed")
+var errInvalidRequest = errors.New("invalid query name")
 
 // Services implements the ServiceBackend interface.
 func (k *Kubernetes) Services(state request.Request, exact bool, opt middleware.Options) ([]msg.Service, []msg.Service, error) {
+
+	if state.Type() == "SRV" && !validSrvPrefix(state.QName()) {
+		return nil, nil, errInvalidRequest
+	}
+
 	s, e := k.Records(state.Name(), exact)
 	return s, nil, e // Haven't implemented debug queries yet.
 }
 
+func validSrvPrefix(name string) bool {
+	matched, _ := regexp.MatchString("^_[^\\.]+\\._(tcp|udp|\\*|any)\\.", strings.ToLower(name))
```

Comment:
do a case insensitive match so that you don't have to lowercase name and create garbage in the process.
(I'll write a parser that will do this for all cases - hopefully without allocating any memory)